core/jobs/batch_jobs/manual_voice_artist_name_job.py

```diff
@@ -189,78 +191,72 @@ def get_content_id_mapping_and_voiceovers_count(
-                            lang_code_to_voiceovers.items()):
 
-                        content_id_to_voiceover_mapping[
-                            content_id][lang_code] = voiceover_dict
-                        number_of_voiceovers += 1
+        for state in exploration.states.values():
+            voiceovers_mapping = (
+                state['recorded_voiceovers']['voiceovers_mapping'])
+            for content_id, lang_code_to_voiceovers in (
+                    voiceovers_mapping.items()):
+                for lang_code, voiceover_dict in (
+                        lang_code_to_voiceovers.items()):
 
-            return (content_id_to_voiceover_mapping, number_of_voiceovers)
-        except Exception as e:
-            raise Exception(
-                'Failed to get voiceover mapping for exploration %s.' %
-                exploration.id
-            ) from e
+                    content_id_to_voiceover_mapping[
+                        content_id][lang_code] = voiceover_dict
+                    number_of_voiceovers += 1
+
+        return (content_id_to_voiceover_mapping, number_of_voiceovers)
 
     @classmethod
     def update_content_id_to_voiceovers_mapping(
         cls,
-        voiceover_mapping_1: Dict[str, Dict[str, state_domain.VoiceoverDict]],
-        voiceover_mapping_2: Dict[str, Dict[str, state_domain.VoiceoverDict]],
+        reference_voiceover_mapping: Dict[
+            str, Dict[str, state_domain.VoiceoverDict]],
+        voiceover_mapping: Dict[
+            str, Dict[str, state_domain.VoiceoverDict]],
         voiceover_artist_and_voiceover_mapping: (
             voiceover_domain.ContentIdToVoiceoverMappingType),
         voice_artist_id: str
     ) -> Tuple[voiceover_domain.ContentIdToVoiceoverMappingType, int]:
         """Adds new voiceover entries to the
         `voiceover_artist_and_voiceover_mapping` dictionary with their
         respective voice artists. Only include voiceovers that are present in
-        both the `voiceover_mapping_1` and `voiceover_mapping_2` dictionaries.
+        both the `reference_voiceover_mapping` and `voiceover_mapping`
+        dictionaries.
 
         Args:
-            voiceover_mapping_1: dict(str, dict(str, VoiceoverDict)). The
-                dictionary contains voiceover data for some version of
+            reference_voiceover_mapping: dict(str, dict(str, VoiceoverDict)).
+                The dictionary contains voiceover data for some version of
                 exploration, say N. This dict maps content IDs as keys and
                 nested dicts as values. Each nested dict contains language codes
                 as keys and voiceover dicts as values.
-            voiceover_mapping_2: dict(str, dict(str, VoiceoverDict)). The
+            voiceover_mapping: dict(str, dict(str, VoiceoverDict)). The
                 dictionary contains voiceover data for some other version of
                 exploration, say M. This dict maps content IDs as keys and
```

Comment:
Maybe change "M" to "M < N" (if that's true).

Reply:
Yes, that's true. Updated.